packages/pyright-internal/src/backgroundAnalysis.ts

```diff
@@ -19,18 +19,22 @@ import { ServiceProvider } from './common/serviceProvider';
 import { getRootUri } from './common/uri/uriUtils';
 
 export class BackgroundAnalysis extends BackgroundAnalysisBase {
-    constructor(serviceProvider: ServiceProvider) {
+    constructor(serviceProvider: ServiceProvider, workerIndex = BackgroundThreadIndex) {
```

Comment:
This seems fragile. If more than one BackgroundAnalysis object is instantiated and the caller forgets to pass in a unique index, the two will now step on each other's data in the shared buffer. Is there a way to make this more robust, like dynamically generating a unique ID or index during construction of the BackgroundAnalysis?

Reply:
Oh good point. I'll generate it from a static. This is always called on the main thread.